### ISSUE TITLE
Excessive CPU load

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#4734]
There are complaints about high CPU load which deserve a separate issue, see here https://github.com/osmandapp/Osmand/issues/3898#issuecomment-343414230 and the investigations in the comments above that post.

**Comment by vshcherb** [comment]
Do we have any reference good builds i.e. was it always permanent issue? Or it happened in particular release.

**Comment by CJTmmr** [comment]
from #3898:

> I have seen the high CPU-use as well on Samsung Galaxy Camera 2 (in latest nightlies) In my experience: after a fresh start the CPU-use stays high for a long time, even when OsmAnd is no longer on foreground.The Street name and Lane marker are not shown in follow-me-mode. But even when the map is on a fixed position the CPU-use stays high.I guess the disappearing streetname is a consequence of the high CPU-use, rather than a cause.Only after a search for address or POI is completed, the CPU-use goes back to normal...Is it possible that at start-up some initial search-proces is started, which somehow does not end?Good luck investigating.

Also when a "reverse geo lookup" (e.g. after short tap on POI on the map, or long tap on any location on the map) is completed, CPU-use goes back to almost 0% and streetname/lanes are displayed correctly.
This first geo lookup takes quite some time though - sometimes upto 2 or 3 minutes...

**Comment by scaidermern** [comment]
Please reopen. Commit 9be3164 fixes #4785 but not #4734.

**Comment by vshcherb** [comment]
Since new index for search was introduced, I think the search became much faster so this issue might be obsolete
